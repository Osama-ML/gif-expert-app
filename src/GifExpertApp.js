import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories= {setCategories}/>
            <hr />
            
            <ul>
                {
                    categories.map(categoria => {
                        return <GifGrid 
                            key={categoria}
                            category={categoria}
                        /> ;
                    })
                }
            </ul>
        </>    
    );
}

export default GifExpertApp;