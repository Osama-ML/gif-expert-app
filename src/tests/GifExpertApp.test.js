import React from 'react';
import GifExpertApp from '../GifExpertApp';

import { shallow } from 'enzyme';

describe('Pruebas en GifExpertApp', () => {

    const wrapper = shallow(<GifExpertApp/>)

    test('Debe tener la misma estructura que el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
  
    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['One punch','Dragonball'];
        const wrapper = shallow(<GifExpertApp defaultCategories= {categories}/>)


        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    

})
