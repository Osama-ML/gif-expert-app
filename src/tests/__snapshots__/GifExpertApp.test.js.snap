// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe tener la misma estructura que el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe de mostrar una liste de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragonball"
      key="Dragonball"
    />
  </ul>
</Fragment>
`;
